text/0000-format-args-implicit-identifiers.md

```diff
@@ -0,0 +1,487 @@
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+If this proposal were captured, the following (currently invalid) macro invocation:
+
+    format_args!("hello {person}")
+
+would become a valid macro invocation, and would be equivalent to a shorthand for the already valid:
+
+    format_args!("hello {person}", person=person)
+
+This identifier `person` would be known as an **implicit named argument** to the macro. `format_args!` would be able to accept any number of such implicit named arguments in this fashion. Each implicit named argument would have to be an identifier which existed in the scope in which the macro is invoked.
+
+Should `person` not exist in the scope, the usual error E0425 would be emitted by the compiler:
+
+    error[E0425]: cannot find value `person` in this scope
+     --> .\foo.rs:X:Y
+      |
+    X |     println!("hello {person}");
+      |                     ^^^^^^^^ not found in this scope
+
+As a result of this change, downstream macros based on `format_args!` would also be able to accept implicit named arguments in the same way. This would provide ergonomic benefit to many macros across the ecosystem, including:
+
+ - `format!`
+ - `print!` and `println!`
+ - `eprint!` and `eprintln!`
+ - `write!` and `writeln!`
+ - `panic!`, `unreachable!` and `unimplemented!`
+ - `assert!` and similar
+ - macros in the `log` crate
+ - macros in the `failure` crate
+
+(This is not an exhaustive list of the many macros this would affect. In discussion of this RFC if any further commonly-used macros are noted, they may be added to this list.)
+
+## Precedence
+
+Implicit arguments would have lower precedence than the existing named arguments `format_args!` already accepts. For example, in the example below, the `person` named argument is explicit, and so the `person` variable in the same scope would not be captured:
+
+    let person = "Charlie";
+
+    // Person is an explicit named argument, so this
+    // expands to "hello Snoopy".
+    println!("hello {person}", person="Snoopy");
+
+Indeed, in this example above the `person` variable would be unused, and so in this case the unused varible warning will apply, like the below:
+
+    warning: unused variable: `person`
+     --> src/foo.rs:X:Y
+      |
+    X |     let person = "Charlie";
+      |         ^^^^^^ help: consider prefixing with an underscore: `_person`
+      |
+      = note: `#[warn(unused_variables)]` on by default
+
+Because implicit named arguments would have lower precedence than explicit named arguments, it is anticipated that no breaking changes would occur to existing code by implementing this RFC.
+
+## Generated Format Strings
+
+`format_args!` can accept an expression instead of a string literal as its first argument. `format_args!` will attempt to expand any such expression to a string literal. If successful then the `format_args!` expansion will continue as if the user had passed that string literal verbatim.
```

Comment:
Is this the description of an existing behavior, or a new feature proposed by this RFC? If the former, where can I read more about this?

Reply:
Never mind :) I found the Macro Hygiene section which explains in more details, and https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e42db7da7dba251aab15ccfa8ead3b65 shows that this is existing behavior.